application/models/Question.php

```diff
@@ -726,54 +726,94 @@ public function getOrderedAnswers($scale_id = null)
-        if ($alphasort == 1) {
-            foreach ($aAnswerOptions as $scaleId => $aScaleArray) {
-                $aSorted = array();
+        // Sort alphabetically if applicable
+        if ($this->shouldOrderAnswersAlphabetically()) {
+            foreach ($answerOptions as $scaleId => $scaleArray) {
+                $sorted = array();
 
-                // We create an aray aSorted that will use the answer in the current language as key, and that will store its old index as value
-                foreach ($aScaleArray as $iKey => $oAnswer) {
-                    $aSorted[$oAnswer->answerl10ns[$this->survey->language]->answer] = $iKey;
+                // We create an aray sorted that will use the answer in the current language as key, and that will store its old index as value
+                foreach ($scaleArray as $key => $answer) {
+                    $sorted[$answer->answerl10ns[$this->survey->language]->answer] = $key;
                 }
+                ksort($sorted);
 
-                ksort($aSorted);
-
-                // Now, we create a new array that store the old values of $aAnswerOptions in the order of $aSorted
-                $aNew = array();
-                foreach ($aSorted as $sAnswer => $iKey) {
-                    $aNew[] = $aScaleArray[$iKey];
+                // Now, we create a new array that store the old values of $answerOptions in the order of $sorted
+                $sortedScaleAnswers = array();
+                foreach ($sorted as $answer => $key) {
+                    $sortedScaleAnswers[] = $scaleArray[$key];
                 }
-                $aAnswerOptions[$scaleId] = $aNew;
+                $answerOptions[$scaleId] = $sortedScaleAnswers;
             }
-            return $aAnswerOptions;
+            return $answerOptions;
         }
 
-        foreach ($aAnswerOptions as $scaleId => $aScaleArray) {
-            usort($aScaleArray, function ($a, $b) {
+        // Sort by Answer's own sort order
+        foreach ($answerOptions as $scaleId => $scaleArray) {
+            usort($scaleArray, function ($a, $b) {
                 return $a->sortorder > $b->sortorder
                     ? 1
                     : ($a->sortorder < $b->sortorder ? -1 : 0);
             });
-            $aAnswerOptions[$scaleId] = $aScaleArray;
+            $answerOptions[$scaleId] = $scaleArray;
         }
-        return $aAnswerOptions;
+        return $answerOptions;
+    }
+
+    /**
+     * Returns true if the answer options should be ordered randomly.
+     * @return bool
+     */
+    private function shouldOrderAnswersRandomly()
+    {
+        // Question types supporting both Random Order and Alphabetical Order should
+        // implement the 'answer_order' attribute instead of using separate attributes.
+        $answerOrder = $this->getQuestionAttribute('answer_order');
+        if (!is_null($answerOrder)) {
+            return $answerOrder == 'random';
```

Comment:
Not used after

Reply:
Sorry, don't understand

Reply:
`$answerOrder` variable is not used in function 

The function do ` return $this->getQuestionAttribute('random_order') == 1 && $this->getQuestionType()->subquestions == 0;`

Reply:
> $answerOrder variable is not used in function

It is used in that snippet that you comment. Don't understand.

> The function do return $this->getQuestionAttribute('random_order') == 1 && $this->getQuestionType()->subquestions == 0;

That's later, in case the answerOrder approach doesn't go well (ex: questions which don't have the new attribute)

Reply:
Argl … didn't see the "return"

I muts understand the reason of this update :).